doc/user_guide/interactions/parameters.rst

```diff
@@ -428,41 +508,154 @@ cylinders:
+~~~~~~~~~~~~~~~~~~~~
+
+Multiple parameters can be combined in a single chart,
+either via multiple separate response conditions,
+different conditional branches in :fun:`when`,
+or parameter composition.
+
+Multiple conditions
+^^^^^^^^^^^^^^^^^^^
+
+In this example,
+points that are hovered with the pointer
+will increase in size
+and those that are clicked
+will be filled in with red.
+The ``empty=False`` is to ensure that no points are selected to start.
+Try holding shift to select multiple points on either hover or click.
+
+.. altair-plot::
+
+    click = alt.selection_point(empty=False)
+    hover = alt.selection_point(on='pointerover', empty=False)
+
+    points = alt.Chart(cars).mark_point().encode(
+        x='Horsepower:Q',
+        y='Miles_per_Gallon:Q',
+        fill=alt.when(click).then(alt.value('red')),
+        size=alt.when(hover).then(alt.value(1000))
+    ).add_params(
+        click, hover
+    )
+
+    points
+
+.. _conditional-branches:
+
+Conditional branches
+^^^^^^^^^^^^^^^^^^^^
+
+:fun:`when` allows the use of multiple ``then`` (``elif``) branches
+which can change the behavior of a single encoding
+in response to multiple different parameters.
+Here,
+we fill hovered points in yellow,
+before changing the fill to red
+when a point is clicked.
+Since the mouse is hovering over points
+while clicking them,
+both conditions will be active
+and the earlier branch takes precedence
+(you can try by changing the order of the two ``when.then`` clauses
+and observing that the points will not change to red when clicked).
+
+.. altair-plot::
+
+    click = alt.selection_point(empty=False)
+    hover = alt.selection_point(on='pointerover', empty=False)
+
+    points = alt.Chart(cars).mark_point().encode(
+        x='Horsepower:Q',
+        y='Miles_per_Gallon:Q',
+        fill=(
+            alt.when(click)
+            .then(alt.value('red'))
+            .when(hover)
+            .then(alt.value('gold'))
+        ),
+        size=alt.when(hover).then(alt.value(1000))
+    ).add_params(
+        click, hover
+    )
+
+    points
+
+.. _parameter-composition:
 
 Parameter Composition
-~~~~~~~~~~~~~~~~~~~~~
+^^^^^^^^^^^^^^^^^^^^^
 
 Altair also supports combining multiple parameters using the ``&``, ``|``
 and ``~`` for respectively ``AND``, ``OR`` and ``NOT`` logical composition
 operands.
+These parameter compositions can be used with both filters and conditions in Altair.
+In the following example,
+only the points that fall within the interval selections of both the scatter plots
+will be counted in the bar chart
+(so you will need to make a selection in both charts
+before the bars shows up).
 
-Returning to our heatmap examples,
-we can construct a scenario where there are two people who can make an interval
+.. altair-plot::
+
+    # empty=False ensure that no points are selected before a selection is drawn
+    brush = alt.selection_interval(empty=False)
+    brush2 = alt.selection_interval(empty=False)
+
```

Comment:
Note that this and the next example will not quite work as intended until https://github.com/vega/altair/issues/3598 is fixed, but I don't think that is blocking for this PR.

Reply:
Good point, I added back the old example and will put this up as a separate PR.